docs/UndoRedo.md

```diff
@@ -0,0 +1,184 @@
+# Undo/Redo
+
+The undo/redo functionality allows the user to revert changes made while editing the dashboard. 
+
+## Properties
+
+### `State`
+
+The dashboard state stores all necessary information to render dashboard. 
+
+Type: Object
+
+A state contains the following fields:
+
+ - `dashboardConfiguration`
+
+    Holds the array of widgets as well as viewport information.
+
+    Type: DashboardConfiguration
+
+ - `selectedWidgetIds`
+
+    The ids of the widgets selected.
+
+    Type: string[]     
+
+ - `numTimesCopyGroupHasBeenPasted`
+
+    The number of times the copy group has been pasted. 
+
+    Type: number
+
+- `copyGroup`
+
+    The group of widgets that have been copied. 
+
+    Type: Widget[]
+
+- `stretchToFit`
+
+    Holds whether or not the dashboard should be in stretch to fit mode. 
+
+    Type: boolean
+
+- `width`
+
+    The width in pixels of dashboard. 
+
+    Type: number
+
+- `cell size`
+
+    The cell size in pixels of dashboard. 
+
+    Type: number
+
+- `intermediateDashboardConfiguration`
+
+    The dashboard configuration while an action is taking place, such as a widget being moved or resized. 
+
+    Type: DashboardConfiguration
+
+- `undoQueue`
+
+    Queue that holds all previous actions perfomed.
+
+    Type: DashboardAction[]
+
+- `redoQueue`
+
+    Queue that holds all previous actions undone.
+
+    Type: DashboardAction[]
+
+- `previousPosition`
+
+    The position used in move as reference for where move started.     
+
+    Type: Position
+
+#### Example
+```
+const state: DashboardState = {
+    dashboardConfiguration: dashboardConfig,
+    selectedWidgetIds: [MOCK_KPI_WIDGET],
+    numTimesCopyGroupHasBeenPasted: 0,
+    copyGroup: [MOCK_KPI_WIDGET],
+    stretchToFit: DEFAULT_STRETCH_TO_FIT,
+    width: DEFAULT_WIDTH,
+    cellSize: DEFAULT_CELL_SIZE,
+    intermediateDashboardConfiguration: undefined,
+    undoQueue: [onMoveAction({position: {x: 1,y:1}, isActionComplete: True, dashboardConfiguration})],
+    redoQueue: [],
+    previousPosition: undefined,
+  };
+```
+
+### `Actions`
```

Comment:
What are all the actions that are responsible for undoing and redoing? They could be documented here.